### PR TITLE
Guesses

### PR DESCRIPTION
**Pull request by marijana777**
Added new files.

### CODE REVIEW

**Comment by profxj** [comment]
Here is a first comment:

I worry about your naming system for the abs_system files (and to a lesser
extend the guesses files).  Would be good to have JXXXX+XXXX_z#.###
for the abs_systems so you know more precisely the QSO and you also
have the redshift.

You can use git mv 
to rename files.

**Comment by profxj** [comment]
I suggest you move the hstselect file from docs/
to data/

**Comment by profxj** [comment]
Notebooks are not code.  And you don't need a code/ folder
under cos_lrg either.  It already is code.

I suggest you move the one Notebook to docs/nb

File: cos_lrg/lrgew.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# import
+import pdb
+import numpy as np
+import glob
+jj
+from linetools.spectra import io as lsio
+from pyigm.abssys.lls import LLSSystem
+
+
+def measure_ew(xabsspath,contflspath, do_aodm=False, verbose=True):
```
Comment (profxj) [inline cos_lrg/lrgew.py]: I like that this is in a module now, but you might want to turn
it into a script.  I guess it is fine for now.

File: cos_lrg/lrgew.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    """ Writing EWs to JSON file
+
+    Parameters
+    ----------
+    xabsspath: str
```
Comment (profxj) [inline cos_lrg/lrgew.py]: xabsspath and contflspath should have default values.

I suggest you do:

```
from pkg_resources import resource_filename

xabsspth = resource_filename('cos_lrg', 'data/lrg_xabssys')
```

File: cos_lrg/lrgew.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    
+    Returns
+    -------
+
+    """
+
+    xabssfiles=glob.glob(xabsspath+'*')
+    contfls=glob.glob(contflspath+'*')
+    #sort / match
+    xabssfiles=np.sort(xabssfiles)
```
Comment (profxj) [inline cos_lrg/lrgew.py]: this is too risky.

You should use your summary file as the starting point
for all such analysis (i.e hstselect_final.fits) and use
it to loop on the LRGs and then search for the abssystem
and the spectrum.

You should make a new module named io.py
to deal with loading spectra and AbsSystems etc.

File: cos_lrg/lrgew.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                absline.measure_aodm()
+
+            if verbose:
+                print(absline)
+                print('EW = {:g} with error {:g}'.format(absline.attrib['EW'],absline.attrib['sig_EW']))
+                print(' ')
+        if verbose:
+            print('====================================')
+            print('   ')
+        # Write updated JSON file to disk
```
Comment (profxj) [inline cos_lrg/lrgew.py]: You need to add code to write the updated AbsSystem
to disk.

File: cos_lrg/lrgew.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    xabssfiles=np.sort(xabssfiles)
+    contfls=np.sort(contfls) 
+    
+    #for i in np.arange(len(xabssfiles)):
+    #    print(xabssfiles[i],contfls[i])
+    
+    for i in np.arange(len(xabssfiles)):
+        if verbose:
+            print(xabssfiles[i])
+        lls = LLSSystem.from_json(xabssfiles[i], chk_vel=False)
```
Comment (profxj) [inline cos_lrg/lrgew.py]: I doubt that all of your LRGs have LLS.

You should probably use GenericAbsSystem or whatever it is called.

File: cos_lrg/setup_package.py [hunk 6]
```diff
@@ -1,5 +1,51 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
```
Comment (profxj) [inline cos_lrg/setup_package.py]: This is a special file.  Please return it to the way it was.

You can put your new code in a file like io.py

Thanks

File: cos_lrg/setup_package.py [hunk 7]
```diff
@@ -1,5 +1,51 @@
...
+    # read fits table
+    datainfo = Table.read(datafile)
+    coords = ltu.radec_to_coord((datainfo['RA_QSO'], datainfo['DEC_QSO']))
+    names = []
+    for coord in coords:
+        name = 'J{:s}{:s}'.format(coord.ra.to_string(unit=u.hour,sep='',pad=True, precision=2),coord.dec.to_string(sep='',pad=True,alwayssign=True, precision=1))
+        names.append(name)
+    datainfo['NAME'] = names
+            
+    for i in np.arange(len(datainfo)):
```
Comment (profxj) [inline cos_lrg/setup_package.py]: For loops are a bit painful to use and painful to look at.

You could and should do a search on coordinates or 
simple bool logic.  But this is fine for now.

File: cos_lrg/setup_package.py [hunk 8]
```diff
@@ -1,5 +1,51 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 
-def get_package_data():
-    return {'cos_lrg.tests': ['files/*']}
+#def get_package_data():
+#    return {'cos_lrg.tests': ['files/*']}
+
+def get_package_data(datafile,path_to_files):
```
Comment (profxj) [inline cos_lrg/setup_package.py]: in your new module, rename this to something
sensible.

**Comment by profxj** [comment]
@marijana777 -- somehow you accidentally added a whole bunch of files
that we don't want in the repo.  Please git rm all the files in:

.eggs
.idea
COSLRGs.egg-info

those are made when one installs Python
and having them in here will cause conflicts.

File: cos_lrg/scripts/make_systems.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    if args.lrg_name == 'ALL':
+        pdb.set_trace() # NOT READY FOR THIS YET
+    else:
+        icoords = [get_coord(args.lrg_name)]
+
+    lrgs_sys = []
+
+    for icoord in icoords:
+        # Load the Spectrum
+        spec = load_spectrum(icoord)
```
Comment (profxj) [inline cos_lrg/scripts/make_systems.py]: Do you need to load the spectrum?

File: cos_lrg/scripts/make_systems.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        row = match_coord_to_summ(icoord)
+        zlrg = row['Z_GAL']
+        # load guesses file and systems
+        igm_systems, guesses_file = load_guesses(icoord)
+
+        # find lrg system(s?)
+        dzabssys = []
```
Comment (profxj) [inline cos_lrg/scripts/make_systems.py]: An altenate way to code this with a list iterator:

```
z_sys = [igm_sys.zabs for igm_sys in igm_systems]
dzabssys = np.abs(np.array(z_sys) - zlrg))
```

File: cos_lrg/scripts/make_systems.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+        for igm_sys in igm_systems:   ## igm_sys[0].zabs
+            dzabssys.append(np.abs(igm_sys.zabs - zlrg))
+        idx = np.argmin(dzabssys)
```
Comment (profxj) [inline cos_lrg/scripts/make_systems.py]: I suggest you put in some tolearance here, e.g.
the system should lie within XXXkm/s of zlrg

File: cos_lrg/scripts/make_systems.py [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+        lrgsys = igm_systems[idx]
+
+        # write lrg system in a file
+
+        # new folder
+        lent = len(guesses_file.split('/')[-1]) + len(guesses_file.split('/')[-2]) + 1
```
Comment (profxj) [inline cos_lrg/scripts/make_systems.py]: Not sure I understand the new folder. 
Are you making one for each LRG?
And where in the Repo?

File: cos_lrg/scripts/make_systems.py [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+        dec = coord.dec.to_string(sep='', pad=True, alwayssign=True, precision=1)[0:5]
+        # Full file
+        lrg_g_file = 'LRG_guesses_J{:s}{:s}_z{:0.3f}.json'.format(ra, dec, zlrg)
+        lrg_g_file_full = lrg_g_folder+lrg_g_file
+        print("Writing/Overwriting file {:s} ?".format(lrg_g_file_full))
+        pdb.set_trace()
+        lrgsys.write_json(outfil=lrg_g_file_full)
+        lrgs_sys.append(lrgsys)
+
+    return lrgs_sys
```
Comment (profxj) [inline cos_lrg/scripts/make_systems.py]: No need to return something in a script like this

File: cos_lrg/scripts/make_systems.py (hunk 12 above)
Comment (marijana777) [inline cos_lrg/scripts/make_systems.py]: One folder for all LRGs. 
guesses_file = '/user/COSLRGs/cos_lrg/data/guesses/J0226_igmgs.json'
lent = len(guesses_file.split('/')[-1]) + len(guesses_file.split('/')[-2]) + 1
lrg_g_folder = guesses_file[0:-lent] + 'lrg_from_guesses/'
lrg_g_folder

'/user/COSLRGs/cos_lrg/data/lrg_from_guesses/'

**Pull request merged**